android/app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java

```diff
@@ -169,6 +176,22 @@ private void setToolbar() {
             getSupportActionBar().setElevation(Utils.dpToPx(4));
     }
 
+    private void updateEventDates() {
+        Event event = RealmDataRepository.getDefaultInstance().getEventSync();
+
+        String startTime = event.getStartsAt();
+        String endTime = event.getEndsAt();
+
+        Observable.fromCallable(() ->
+                DateConverter.getDaysInBetween(startTime, endTime)
+        ).subscribe(eventDates -> RealmDataRepository.getDefaultInstance().saveEventDates(eventDates).subscribe(), throwable -> {
+            Timber.e(throwable);
+            Timber.e("Error updating event dates : enable to parse start date: %s and end date: %s in ISO format",
+                    startTime, endTime);
+            OpenEventApp.postEventOnUIThread(new RetrofitError(new Throwable("Error parsing dates")));
+        });
+    }
```

Comment:
This whole block is synchronous, even the fromCallable method. So, there is no need for the fromCallable method.